[FIX] graphql_base: pin graphql-server <3.0.0

Mohajiro opened the pull request:
graphql-server 3.0.0 removed HttpQueryError, which breaks graphql_base controller on Odoo 16. 
Pin dependency to <3.0.0 to keep stable behavior.

[Ref: OCA/rest-framework#559](https://github.com/OCA/rest-framework/issues/559)

Task: 4927

Mohajiro committed:
[FIX] graphql_base: pin graphql-server <3.0.0

graphql-server 3.0.0 removed HttpQueryError, which breaks
graphql_base controller on Odoo 16. Pin dependency to <3.0.0
to keep stable behavior.

Task: 4927

sbidoul commented:
Thanks! This looks good. Can you remove the change to `index.html`, to make ports easier?

sbidoul commented on graphql_base/__manifest__.py:
```diff
@@ -11,7 +11,7 @@
     "depends": ["base"],
     "data": ["views/graphiql.xml"],
-    "external_dependencies": {"python": ["graphene", "graphql_server"]},
+    "external_dependencies": {"python": ["graphene", "graphql-server<3.0.0b8"]},
```
```suggestion
    # We place an upper bound on graphql-server because it has undergone a major
    # change in 3.0.0b8 in Aug 2025. When graphql-server 3 has stabilized we'll need
    # to update this module to make it compatible with it.
    "external_dependencies": {"python": ["graphene", "graphql-server<3.0.0b8"]},
```

Mohajiro replied:
It's done. It's okey?

Mohajiro committed:
[FIX] graphql_base: restore index and comments in manifest

Add comments im manifest.
Restore initial version of index.html to make ports easier.

Task: 4927

ivs-cetmix reviewed (approved):
Code review LGTM

ivs-cetmix commented:
Hi @sbidoul , could you please have a look at this at your earliest convenience?

Aldeigja reviewed (approved):
Functional LGTM

yvaucher commented:
/ocabot merge patch
